client/homebrew/editor/stringArrayEditor/stringArrayEditor.jsx

```diff
@@ -109,35 +126,64 @@ const StringArrayEditor = createClass({
 		this.setState({ valueContext, updateValue: '' });
 	},
 
+	handleTagKeyDown : function(event, index) {
+		if(_.includes(['Enter', 'Space'], event.code)){
+			event.preventDefault();
+			this.editValue(index);
+			setTimeout(()=>{
+				event.target.querySelector('input').focus();
+			}, 0);
+		} else if(_.includes(['Delete'], event.key)){
+			this.removeValue(index);
+		} else if(_.includes(['ArrowLeft'], event.key)){
+			event.target.previousElementSibling?.focus();
+		} else if(_.includes(['ArrowRight'], event.key)){
+			event.target.nextElementSibling?.focus();
+		}
+
+
+	},
+
+	renderDatalist : function(){
+		if(this.props.options?.length > 0){
+			return <datalist id={`${this.props.id}__tags-precoordinated`}>
+				{this.props.options.map((option)=>{
+					return <option value={`${option}`}></option>;
+				})}
+			</datalist>;
```

Comment:
Is it possible to display a different value in the datalist options text? Just thinking, it may be nice to show something like `meta: ...` to indicate that `meta:` by itself isn't a valid option. Then when you click, it just applies `meta:` and keeps the datalist open to now show the suggestions to fill out the tag.

Or instead of `...`, the label shows some example sub-tags to explain what is meant:

Edit: Seems you can add text to the `<option>` element to display sub-text:

```
<option value="type:">(adventure, class, setting, etc.)</option>
```

![image](https://github.com/naturalcrit/homebrewery/assets/5878534/442d5c1d-5e3d-44ed-b5a1-6ebc5c51aef5)

Maybe combine this with my earlier comment to show a simplified list when the textbox is first empty so they just see a high-level list, but then use the full list after the user starts typing.

Reply:
At some point, would it be worth it to tweak our existing dropdown menu thing so we have full customizability?